wsmaster/che-core-api-user/src/main/java/org/eclipse/che/api/user/server/model/impl/ProfileImpl.java

```diff
@@ -21,19 +31,36 @@
  *
  * @author Yevhenii Voevodin
  */
+@Entity(name = "Profile")
 public class ProfileImpl implements Profile {
 
-    private String              id;
-    private String              email;
+    @Id
+    @Column(name = "user_id")
+    private String userId;
+
+    @PrimaryKeyJoinColumn
+    public UserImpl user;
+
+    @ElementCollection
+    @MapKeyColumn(name = "name")
+    @Column(name = "value")
+    @CollectionTable(joinColumns = @JoinColumn(name = "user_id"),
+                     uniqueConstraints = @UniqueConstraint(columnNames = {"user_id", "name"}))
     private Map<String, String> attributes;
 
-    public ProfileImpl(String id, String email) {
-        this.id = id;
+    // jpa mapping delegated to 'user' field
+    @Transient
```

Comment:
Email is stored by `User` table while it is used in profile as well, there is no need to store email so the mapping is delegated to the User's field, on the other hand we need to create new profile instances from another profile instances or any data, in this way we need email field, which allows to avoid creation of embedded user object.